### PR TITLE
Run phase for competition

### PR DESCRIPTION
**Pull request by Peque**
Fingers crossed... :crossed_fingers:  :joy:

### CODE REVIEW

**Comment by Peque** [comment]
@cua-cua I think it should be working now. :sweat_smile:

File: src/main.c [hunk 1]
```diff
@@ -30,38 +29,15 @@ void sys_tick_handler(void)
...
-	stop_middle();
-	solved = true;
-}
-
 /**
  * @brief Includes the user configurations.
  */
-static void user_configuration(void)
+static void user_configuration(bool run)
 {
```
[the remark this answers is not in the document]
Comment (cua-cua) [reply src/main.c]: The user configuration is always used, but the available modes change (maybe include it on the brief?)

File: src/target/move.c [hunk 2]
```diff
@@ -310,7 +324,18 @@ void move_front(void)
...
 	enable_walls_control();
 	target_straight(current_cell_start_micrometers, CELL_DIMENSION,
-			get_max_linear_speed());
+			get_max_end_linear_speed());
+	entered_next_cell();
+}
+
+/**
+ * @brief Move front many cells.
+ */
```
Comment (cua-cua) [inline src/target/move.c]: Include input parameter cells

**Comment by cua-cua** [comment]
I will use the current opened #254 to review in general the functions and try to get system state easily.

File: src/main.c [hunk 3]
```diff
@@ -138,12 +117,15 @@ int main(void)
...
+	add_goal(3, 5);
+	set_target_goal();
 	initialize_solver_direction();
-	set_speed_mode(0);
+	set_speed_mode(0, false);
 	while (1) {
 		if (button_left_read_consecutive(500)) {
 			before_moving();
-			solve();
+			explore();
```
[the remark this answers is not in the document]
Comment (Peque) [reply src/main.c]: Yeah, this was due to refactoring, after trying to simplify the API and make it more understandable. The name "solve" was a bit unclear maybe, so now there are "explore" and "run", which are more understandable.

File: src/main.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Peque) [reply src/main.c]: Fixed in f0ec3a8f921963eb0299c9cb90710ce9e12b0ce3

File: src/main.c [hunk 4]
```diff
@@ -14,7 +14,6 @@
 #include "solve.h"
 #include "speed.h"
 
-static bool solved;
 static void competition(void);
```
[the remark this answers is not in the document]
Comment (Peque) [reply src/main.c]: https://github.com/Theseus/bulebule/issues/302

File: src/target/move.c (hunk 2 above)
Comment (Peque) [reply src/target/move.c]: Fixed in 90f10e1963310b0d98664f00a7d0291355aea7ce

**Comment by Peque** [comment]
@cua-cua Can you re-review after the latest changes? :innocent:

**Pull request merged**